File: scribbleshare-frontend/src/scripts/protocol/client/messages/ClientMessageMouseMove.js
import ClientMessage from "../ClientMessage.js";
import ClientMessageType from "../ClientMessageType.js";

export default class ClientMessageMouseMove extends ClientMessage {
    constructor(mouseMoves) {
        super(ClientMessageType.MOUSE_MOVE);
        this.mouseMoves = mouseMoves;
    }

    serialize(writer) {
        super.serialize(writer);
        writer.writeUint8(this.mouseMoves.length);
        for (let i = 0; i < this.mouseMoves.length; i++) {
            this.mouseMoves[i].serialize(writer);
        }
    }
}

File: scribbleshare-room/src/main/java/net/stzups/scribbleshare/room/server/websocket/protocol/client/messages/ClientMessageMouseMove.java
package net.stzups.scribbleshare.room.server.websocket.protocol.client.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.room.server.websocket.MouseMove;
import net.stzups.scribbleshare.room.server.websocket.protocol.client.ClientMessage;
import net.stzups.scribbleshare.room.server.websocket.protocol.client.ClientMessageType;

public class ClientMessageMouseMove extends ClientMessage {
    private final MouseMove[] mouseMoves;

    public ClientMessageMouseMove(ByteBuf byteBuf) {
        super(ClientMessageType.MOUSE_MOVE);
        mouseMoves = new MouseMove[byteBuf.readUnsignedByte()];
        for (int i = 0; i < mouseMoves.length; i++) {
            mouseMoves[i] = new MouseMove(byteBuf);
        }
    }

    public MouseMove[] getMouseMoves() {
        return mouseMoves;
    }
}
